### PR TITLE
Generate thinner .env files in generate_tests.py, switch to GKE deployment

### PR DESCRIPTION
**Pull request by zmerlynn**
This commit does the following:
* Takes advantage of --deployment=gke, --ginkgo-parallel, etc.
* Adds an --env= to kubenetes_e2e.py and takes advantage of it in
generated tests. We're getting to the point where there's only a
couple of envs, and it makes config.json arguably easier to understand.
* Doesn't emit (and attempts to remove) tests where the env is now empty.

### CODE REVIEW

**Comment by zmerlynn** [comment]
cc @fejta @krzyzacy

File: experiment/test_config.yaml [hunk 1]
```diff
@@ -660,29 +656,22 @@ testSuites:
     args:
     - --timeout=500m
     - --test_args=--ginkgo.focus=\[Serial\]|\[Disruptive\] --ginkgo.skip=\[Flaky\]|\[Feature:.+\] --minStartupPods=8
-    envs:
-    - GINKGO_PARALLEL=n
+    - --ginkgo-parallel
```
Comment (yguo0905) [inline experiment/test_config.yaml]: `--ginkgo-parallel=1`?

Comment (zmerlynn) [inline experiment/test_config.yaml]: Good catch!

Comment (zmerlynn) [inline experiment/test_config.yaml]: Fixed, PTAL

**Comment by yguo0905** [comment]
/lgtm

Thanks for eliminating the env files! Much cleaner now.

**Pull request merged**